src/pthread-main.js

```diff
@@ -52,19 +52,39 @@ Module['print'] = threadPrint;
 Module['printErr'] = threadPrintErr;
 this.alert = threadAlert;
 
+Module['instantiateWasm'] = function(info, receiveInstance) {
+  // Instantiate from the module posted from the main thread.
+  // We can just use sync instantiation in the worker.
+  instance = new WebAssembly.Instance(Module['wasmModule'], info);
+  // We don't need the module anymore; new threads will be spawned from the main thread.
+  // XXX is this true?
+  Module['wasmModule'] = undefined;
+  receiveInstance(instance);
+  return instance.exports;
+}
```

Comment:
The whole block above should be `#if WASM` - there is still a short window that I want to retain both asm.js and wasm multithreading, so let's gate this accordingly.